tinyyaml.lua

```diff
@@ -146,11 +146,58 @@ local function countindent(line)
   return j, ssub(line, j+1)
 end
 
-local function parsestring(line, stopper)
+local function isemptyline(line)
+  return line == '' or sfind(line, '^%s*$') or sfind(line, '^%s*#')
+end
+
+local function equalsline(line, needle)
+  return startswith(line, needle) and isemptyline(ssub(line, #needle+1))
+end
+
+local function compactifyemptylines(lines)
```

Comment:
Does it require to move these functions above?

Reply:
Nope, mentioned this is in the comment above. If you prefer without I can remove that move.

Reply:
I just move these functions down:
```diff
diff --git tinyyaml.lua tinyyaml.lua
index 3bf87ec..afa9b90 100644
--- tinyyaml.lua
+++ tinyyaml.lua
@@ -146,49 +146,6 @@ local function countindent(line)
   return j, ssub(line, j+1)
 end

-local function isemptyline(line)
-  return line == '' or sfind(line, '^%s*$') or sfind(line, '^%s*#')
-end
-
-local function equalsline(line, needle)
-  return startswith(line, needle) and isemptyline(ssub(line, #needle+1))
-end
-
-local function compactifyemptylines(lines)
-  -- Appends empty lines as "\n" to the end of the nearest preceding non-empty line
-  local compactified = {}
-  local lastline = {}
-  for i = 1, #lines do
-    local line = lines[i]
-    if isemptyline(line) then
-      if #compactified > 0 and i < #lines then
-        tinsert(lastline, "\n")
-      end
-    else
-      if #lastline > 0 then
-        tinsert(compactified, tconcat(lastline, ""))
-      end
-      lastline = {line}
-    end
-  end
-  if #lastline > 0 then
-    tinsert(compactified, tconcat(lastline, ""))
-  end
-  return compactified
-end
-
-local function checkdupekey(map, key)
-  if rawget(map, key) ~= nil then
-    -- print("found a duplicate key '"..key.."' in line: "..line)
-    local suffix = 1
-    while rawget(map, key..'_'..suffix) do
-      suffix = suffix + 1
-    end
-    key = key ..'_'..suffix
-  end
-  return key
-end
-
 local Parser = {
   timestamps=true,-- parse timestamps as objects instead of strings
 }
@@ -268,6 +225,49 @@ function Parser:parsestring(line, stopper)
   return rtrim(buf), line
 end

+local function isemptyline(line)
+  return line == '' or sfind(line, '^%s*$') or sfind(line, '^%s*#')
+end
+
+local function equalsline(line, needle)
+  return startswith(line, needle) and isemptyline(ssub(line, #needle+1))
+end
+
+local function compactifyemptylines(lines)
+  -- Appends empty lines as "\n" to the end of the nearest preceding non-empty line
+  local compactified = {}
+  local lastline = {}
+  for i = 1, #lines do
+    local line = lines[i]
+    if isemptyline(line) then
+      if #compactified > 0 and i < #lines then
+        tinsert(lastline, "\n")
+      end
+    else
+      if #lastline > 0 then
+        tinsert(compactified, tconcat(lastline, ""))
+      end
+      lastline = {line}
+    end
+  end
+  if #lastline > 0 then
+    tinsert(compactified, tconcat(lastline, ""))
+  end
+  return compactified
+end
+
+local function checkdupekey(map, key)
+  if rawget(map, key) ~= nil then
+    -- print("found a duplicate key '"..key.."' in line: "..line)
+    local suffix = 1
+    while rawget(map, key..'_'..suffix) do
+      suffix = suffix + 1
+    end
+    key = key ..'_'..suffix
+  end
+  return key
+end
+
 function Parser:parseflowstyle(line, lines)
   local stack = {}
   while true do
@@ -829,7 +829,7 @@ end

 local function parse(source, options)
   local options = options or {}
-  parser = setmetatable (options, {__index=Parser})
+  local parser = setmetatable (options, {__index=Parser})
   return parser:parse(source)
 end
```

Feel free to apply the patch. Now the changeset is:
1 file changed, 48 insertions(+), 38 deletions(-)

Reply:
i will edit the first commit to move them down then

Reply:
@spacewander I remove the commit with the moving of the functions and created the parser as local. So it should match your adjustments.